Competition/src/main/cpp/subsystems/Shooter.cpp

```diff
@@ -0,0 +1,185 @@
+    flywheel_lead.SetIdleMode(rev::CANSparkMax::IdleMode::kCoast);
+    turret.SetIdleMode(rev::CANSparkMax::IdleMode::kBrake);
+
+    turret.SetInverted(false);
+
+    flywheel_lead.SetInverted(true);
+
+    flywheel_lead.Follow(rev::CANSparkMax::kFollowerDisabled, false);
+    flywheel_follow.Follow(flywheel_lead, true);
+
+    turret.EnableSoftLimit(rev::CANSparkMax::SoftLimitDirection::kForward, true);
+    turret.SetSoftLimit(rev::CANSparkMax::SoftLimitDirection::kForward, ShooterConstants::limitLeft);
+
+    turret.EnableSoftLimit(rev::CANSparkMax::SoftLimitDirection::kReverse, true);
+    turret.SetSoftLimit(rev::CANSparkMax::SoftLimitDirection::kReverse, ShooterConstants::limitRight);
+
+    pidController.SetP(ShooterConstants::shooterKP);
+    pidController.SetI(ShooterConstants::shooterKI);
+    pidController.SetD(ShooterConstants::shooterKD);
+    pidController.SetIZone(ShooterConstants::shooterKFF);
+    pidController.SetFF(ShooterConstants::shooterKIZ);
+    pidController.SetOutputRange(0, 1);
+
+    resetState(); //reset shooter/encoder state
+}
+
+void Shooter::resetState(){
+    resetEncoder();
+    state.turretState = TurretState::TURRET_DEFAULT;
+    state.hoodState = HoodState::HOOD_DISABLE;
+    state.flywheelState = FlywheelState::FLYWHEEL_DEFAULT;
+}
+
+void Shooter::resetEncoder(){
+    turretEncoder.SetPosition(0);
+    hoodEncoder.SetPosition(0);
+}
+
+void Shooter::assessInputs()
+{
+    if (!operatorController)
+    {
+        return;
+    }
+
+    // driver inputs
+    state.leftStickX = operatorController->GetLeftX();
+    state.startButton = operatorController->GetStartButton();
+    state.backButton = operatorController->GetBackButton(); 
+    //state.dPadDownPressed = driverController->GetPOV(frc::GenericHID::)
+    
+    //Turret
+    if (std::abs(state.leftStickX) > ShooterConstants::kDeadband) {
+        state.turretState = TurretState::TURRET_MANUAL;
+    }
+    else if(state.startButton){
+       state.turretState = TurretState::TURRET_PRIME;
+    }
+    else if(state.backButton){
+        state.turretState = TurretState::TURRET_DEFAULT;
+    }
+
+    //Hood
+    if(state.startButton){
+        state.hoodState = HoodState::HOOD_PRIME;
+    }
+    else if(state.backButton){
+        state.hoodState = HoodState::HOOD_DISABLE;
+    }
+
+    //Flywheel
+    if(state.startButton){
+        state.flywheelState = FlywheelState::FLYWHEEL_PRIME;
+    }
+    else if (state.backButton){
+        state.flywheelState = FlywheelState::FLYWHEEL_DEFAULT;
+    }
+}
+
+void Shooter::analyzeDashboard()
+{
+    if(table->GetBoolean("Home Turret", false)){
+        state.turretState = TurretState::TURRET_HOME;
+    }
+}
+
+void Shooter::assignOutputs()
+{   
+    state.turretTarget = 0;
+
+    //DISABLED
+    if(state.turretState == TurretState::TURRET_DISABLE){
+        state.turretTarget = 0;
+    }//HOME
+    else if(state.turretState == TurretState::TURRET_HOME){
+        // @TODO Motion Profile to turret home position
```

Comment:
Need motion profile code in place. PID constants can be set to 0 for now, and target positions can be randomly assigned as long as a constant is created and can be modified